collection_json.py

```diff
@@ -6,6 +6,59 @@
 __version__ = '0.1.0'
 
 
+class ArrayProperty(object):
+
+    """Property which converts from any enumerable to a typed Array instance."""
+
+    def __init__(self, cls, name):
+        """Constructs typed array property
+
+        :param cls type: the type of objects expected in the array
+        :param name str: name of the property
+        """
+        self.cls = cls
+        self.name = name
+        self.value_attr = "_" + name
+
+    def __get__(self, instance, owner):
+        return getattr(instance, self.value_attr, None)
+
+    def __set__(self, instance, value):
+        if value is None:
+            value = []
+        setattr(instance, self.value_attr, Array(self.cls, self.name, value))
+
+
+class TypedProperty(object):
+
+    """Property which is assignable only to a specific type of instance.
```

Comment:
I think I prefer "Property which casts automatically to type on assignment." (reads a bit easier to me)